rust/wal3/src/manifest.rs

```diff
@@ -740,6 +754,152 @@ impl Manifest {
             (None, None) => self.initial_offset.unwrap_or(LogPosition::from_offset(1)),
         }
     }
+
+    /// Apply the destructive operation specified by the Garbage struct.
+    #[allow(clippy::result_large_err)]
+    pub fn apply_garbage(&self, garbage: Garbage) -> Result<Self, ScrubError> {
+        let Garbage {
+            dropped_setsum,
+            actions,
+            cutoff: _,
+        } = garbage;
+        let mut new = self.clone();
+        for action in actions {
```

Comment:
when I read this method for the first time, I was expecting it to apply garbage recursively to all the snapshots also but it does not. Intuitively feels like something is off with the API